### ISSUE TITLE
macOS 12 (beta): tailscaled socket doesn't come up

### ISSUE THREAD

**Issue by nervous-inhuman** [tailscale/tailscale#2074]
Tailscale 1.8.5 doesn't seem to work correctly on the new macOS 12 Monterey beta, judging by the connection reset by peer, tailscaled doesn't come up.

```
❯ /Applications/Tailscale.app/Contents/MacOS/Tailscale status
MBDisplay: creating data for display 0x4280F40 (main) with bounds o=[0.000000, 0.000000], s=[1680.000000, 1050.000000]

2021/06/08 09:28:59 Get "http://local-tailscaled.sock/localapi/v0/status": read tcp 127.0.0.1:60061->127.0.0.1:59657: read: connection reset by peer
```

```
❯ /Applications/Tailscale.app/Contents/MacOS/Tailscale version
MBDisplay: creating data for display 0x4280F40 (main) with bounds o=[0.000000, 0.000000], s=[1680.000000, 1050.000000]
1.8.5
  tailscale commit: 118cf0cd75f92bbe3d994a5b1587c9ea3553221b
  other commit: c6fd9c9d5657e82c2d7888804ebce0e8bd6a4556
  go version: go1.16.4-ts1b677aa
```

Output from Console for the "Tailscale.app" 

```
default	09:40:37.045380+0200	Tailscale	Requesting container lookup; personaid = -1, type = NOPERSONA, name = <unknown>, class = 2, identifier = (null), temp = 0, create = 0, euid = 501, uid = 501
default	09:40:37.048793+0200	Tailscale	container_create_or_lookup_for_platform: success
default	09:40:37.106023+0200	Tailscale	CHECKIN: pid=4654
default	09:40:37.126404+0200	Tailscale	CHECKEDIN: pid=4654 asn=0x0-0xb10b1 foreground=0
default	09:40:37.129802+0200	Tailscale	FRONTLOGGING: version 1
default	09:40:37.129853+0200	Tailscale	Registered, pid=4654 ASN=0x0,0xb10b1
default	09:40:37.133271+0200	Tailscale	BringForward: pid=4654 asn=0x0-0xb10b1 bringForward=0 foreground=0 uiElement=1 launchedByLS=1 modifiersCount=1 allDisabled=0
default	09:40:37.133368+0200	Tailscale	BringFrontModifier: pid=4654 asn=0x0-0xb10b1 Modifier 0 hideAfter=0 hideOthers=0 dontMakeFrontmost=0 mouseDown=0/0 seed=0/0
default	09:40:37.134819+0200	Tailscale	Current system appearance, (HLTB: 2), (SLS: 1)
default	09:40:37.139113+0200	Tailscale	No persisted cache on this platform.
default	09:40:37.141166+0200	Tailscale	Failed to copy the SysCfgDict MG key with error: 0
default	09:40:37.144882+0200	Tailscale	Post-registration system appearance: (HLTB: 2)
default	09:40:37.192673+0200	Tailscale	NSApp cache appearance:
-NSRequiresAquaSystemAppearance: 0
-appearance: (null)
-effectiveAppearance: <NSCompositeAppearance: 0x600001a7c800
 (
    "<NSDarkAquaAppearance: 0x600001a7c700>",
    "<NSSystemAppearance: 0x600001a78580>"
)>
default	09:40:37.222093+0200	Tailscale	Shortcuts Contextual Actions enabled
default	09:40:37.233920+0200	Tailscale	Requesting app group container lookup; personaid = -1, type = NOPERSONA, name = <unknown>, identifier = W5364U7YZB.group.io.tailscale.ipn.macos, euid = 501, uid = 501, platform = 1
default	09:40:37.234745+0200	Tailscale	container_create_or_lookup_app_group_path_by_app_group_identifier: success
default	09:40:37.239006+0200	Tailscale	NOGO-LOG: app introduction skipped
default	09:40:37.239056+0200	Tailscale	NOGO-LOG: registerExtension skipping unknown bundleID, assuming AppEx not SysEx
default	09:40:37.334728+0200	Tailscale	NOGO-LOG: BackendClient.create
default	09:40:37.370795+0200	Tailscale	Sending load command to nehelper
default	09:40:37.370901+0200	Tailscale	Clearing 904F45C8-CF46-4E67-A247-C3A68B1AE308 from the loaded configurations
default	09:40:37.376092+0200	Tailscale	Adding 904F45C8-CF46-4E67-A247-C3A68B1AE308 to the loaded configurations
default	09:40:37.450303+0200	Tailscale	Received configuration update from daemon (initial)
default	09:40:37.450865+0200	Tailscale	Current bundle (/Applications/Tailscale.app) does not have a SystemExtensions directory
default	09:40:37.458586+0200	Tailscale	 Saving configuration Tailscale Tunnel with existing signature {length = 20, bytes = 0x14ed3d36f6956512fbce0e76d78a22516468b69a}
default	09:40:37.508122+0200	Tailscale	Received a com.apple.neconfigurationchanged notification with token 56
default	09:40:37.508657+0200	Tailscale	 Successfully saved configuration Tailscale Tunnel
default	09:40:37.509613+0200	Tailscale	Sending load command to nehelper
default	09:40:37.509692+0200	Tailscale	Clearing 904F45C8-CF46-4E67-A247-C3A68B1AE308 from the loaded configurations
default	09:40:37.510493+0200	Tailscale	Adding 904F45C8-CF46-4E67-A247-C3A68B1AE308 to the loaded configurations
default	09:40:37.522305+0200	Tailscale	NOGO-LOG: BackendClient.go; state=<connecting>
default	09:40:37.522355+0200	Tailscale	NOGO-LOG: stopping leftover tunnel from previous run, state=<connecting>
default	09:40:37.537643+0200	Tailscale	NOGO-LOG: tunnel connection status changed to '<disconnecting>'
default	09:40:37.537687+0200	Tailscale	NOGO-LOG: initial restart: polling *not* suppressed
default	09:40:37.538959+0200	Tailscale	NOGO-LOG: Backoff(BackendClient): backoff 13 msec
default	09:40:47.412655+0200	Tailscale	LSExceptions shared instance invalidated for timeout.
default	09:41:02.546281+0200	Tailscale	NOGO-LOG: tunnel connection status changed to '<disconnected>'
default	09:41:02.546664+0200	Tailscale	NOGO-LOG: Backoff(BackendClient): backoff 55 msec
default	09:41:02.614242+0200	Tailscale	NOGO-LOG: tunnel connection status changed to '<connecting>'

```

**Comment by DentonGentry** [comment]
MacOS and iOS both use a Network Extension for the VPN functionality.
iOS 15 developer beta shows a similar symptom: https://github.com/tailscale/tailscale/issues/2072

**Comment by dmkjr** [comment]
Def. following.  Can test if needed.

**Comment by evliu** [comment]
this might be related https://git.zx2c4.com/wireguard-apple/commit/?h=am/tun-fd-ios15&id=cf3ca719ce9605bc8edffd6c68ad61d101e7b9e3

**Comment by dmkjr** [comment]
> this might be related https://git.zx2c4.com/wireguard-apple/commit/?h=am/tun-fd-ios15&id=cf3ca719ce9605bc8edffd6c68ad61d101e7b9e3

Looks like at least one of the issues.  Hopefully they will address soon.  Having a few issues with a few different applications from MacOS.  Glad they release it in "beta" format a good deal in front of it being released for the masses.

**Comment by tiferrei** [comment]
Just wanted to report that since there is a beta for iOS fixing this issue but not for Mac, I figured the issue would be fixed in general on main. As a temp fix I've built Tailscale from main and running it through the CLI, seems to work so far.

**Comment by danderson** [comment]
It is fixed on main, since the affected codepath is common to all Apple OSes. macOS doesn't yet have TestFlight, so it'll be fixed when we release a new version of the app to everyone. Since Apple also doesn't give us any way to roll that back, we're checking that the fix works on iOS first using TestFlight. We'll build an app update for mac soon.

**Comment by Mravuri96** [comment]
Todays release of 1.8.7, has not fixed this issue.

**Comment by the0neyouseek** [comment]
Fixed with the `1.10.0` release

**Comment by nervous-inhuman** [comment]
I can confirm that release [1.10.0](https://github.com/tailscale/tailscale/releases/tag/v1.10.0) fixes the issue. Therefore, I'm closing my ticket.
